### PR TITLE
Handle composable loggers

### LINKED ISSUE #1190: Bad configured winston crashes newrelic 8.11.0
## Description
The application is crashing when uses newrelic with [winston](https://github.com/winstonjs/winston) inside a winston (It doesn't make much sense but it can happen).

Error:
```
~myPath\node_modules\newrelic\lib\instrumentation\winston.js:40
        opts.transports = opts.transports.map((transport) => {
                        ^

TypeError: Cannot set property transports of #<Logger> which has only a getter
    at Object.createWrappedLogger [as createLogger] (~myPath\node_modules\newrelic\lib\instrumentation\winston.js:40:25)
    at Object.<anonymous> (~myPath\main.js:19:31) <------ MY LOGGER WITH BUG
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:79:12)
    at node:internal/main/run_main_module:17:47
```

## Steps to Reproduce
1. Import newrelic
2. Configure and import winston
3. Configure a logger with other winston logger (`const loggerWithBug = winston.createLogger(logger)`)
4. Init project with require newrelic on top

Example:
https://github.com/DenisSlapelis/newrelic-sample

## Your Environment

Node version: 16.6.2
Operating System and version: Windows 10 - Version 21H2 (19044.1706)

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Fixed an issue when creating a winston logger with an already instantiated logger.

## Links
Closes #1190

## Details
We had some missing logic around instrumenting `winston.createLogger`.  You can create a logger from an instianted logger.  When this is the case the transports are read-only and we caused the application to crash. The customer, graciously provide a clear [reproduction case](https://github.com/DenisSlapelis/newrelic-sample).  I added 2 tests around composing loggers and verifying it doesn't create extra log lines in aggregator and more importantly does not crash.

If you comment out the `isStream` bit in instrumentation the tests will fail/crash like bug

```sh
  # Subtest: should not double log nor instrument composed logger
            not ok 1 - Cannot set property transports of \#<Logger> which has only a getter
              ---
              stack: >
                Object.createWrappedLogger [as createLogger]
                (lib/instrumentation/winston.js:57:25)

                Test.<anonymous> (test/versioned/winston/winston.tap.js:232:33)
              at:
                line: 57
                column: 25
                file: lib/instrumentation/winston.js
                function: Object.createWrappedLogger
                method: createLogger
              type: TypeError
              tapCaught: testFunctionThrow
              test: should not double log nor instrument composed logger
              source: >2
                        // combine transport-level formatters with our own
                        opts.transports = opts.transports.map((transport) => {
                ------------------------^
                          if (transport.format) {
                            transport.format = winston.format.combine(transport.format, instrumentedFormatter())
              ...
```

### CODE REVIEW

File: test/versioned/winston/winston.tap.js [hunk 1]
```diff
@@ -48,38 +48,75 @@ tap.test('Winston instrumentation', { bail: true }, (t) => {
     }
   }
 
-  const logStuff = ({ logger, stream }) => {
-    // Log some stuff, both in and out of a transaction
-    logger.info('out of trans')
+  /**
+   * Log lines in and out of a transaction for eveyr logger.
```
Comment (jordigh) [inline test/versioned/winston/winston.tap.js:52]: "eveyr"

File: test/versioned/winston/winston.tap.js [hunk 2]
```diff
@@ -48,38 +48,75 @@ tap.test('Winston instrumentation', { bail: true }, (t) => {
...
+   * @param {Object} opts
+   * @param {DerivedLogger} opts.logger instance of winston
+   * @param {Array} opts.loggers an array of winston loggers
+   * @param {Stream} opts.stream stream used to end test
+   * @param {Test} opts.t tap test
```
Comment (jordigh) [inline test/versioned/winston/winston.tap.js:57]: According to this docstring, shouldn't we also destructure `t`?

File: test/versioned/winston/winston.tap.js [hunk 3]
```diff
@@ -162,6 +199,39 @@ tap.test('Winston instrumentation', { bail: true }, (t) => {
...
+    t.test('should not double log nor instrument composed logger', (t) => {
+      const assertFn = (msg) => {
+        t.equal(msg['entity.name'], undefined, 'should not have entity name')
+        t.equal(msg['entity.type'], undefined, 'should not have entity type')
+        t.equal(msg.timestamp, undefined, 'should not have timestamp as number')
+        t.equal(msg.hostname, undefined, 'should not have hostname as string')
+        t.equal(msg.level, 'info')
+        t.equal(msg['trace.id'], undefined, 'msg should not have trace id')
+        t.equal(msg['span.id'], undefined, 'msg should not have span id')
+        t.ok(msg.message.includes('NR-LINKING'), 'should not contain NR-LINKING metadata')
```
Comment (jordigh) [inline test/versioned/winston/winston.tap.js:212]: `t.ok` but `should not`?

File: test/versioned/winston/winston.tap.js (hunk 2 above)
Comment (bizob2828) [reply test/versioned/winston/winston.tap.js:57]: derp, copy pasta fail. this function doesn't use the test to assert anything

File: test/versioned/winston/winston.tap.js (hunk 3 above)
Comment (bizob2828) [reply test/versioned/winston/winston.tap.js:212]: nice catch, i'll fix the other test that has this as well

**Review by jordigh — APPROVED** [review]

**Pull request merged**